internal/internal_eager_activity.go

```diff
@@ -34,9 +33,7 @@ import (
 type eagerActivityExecutor struct {
 	eagerActivityExecutorOptions
 
-	activityWorker eagerWorker
-	heldSlotCount  int
-	countLock      sync.Mutex
+	activityWorker *baseWorker
```

Comment:
Can we avoid giving `eagerActivityExecutor` the full `activityWorker `? I am not sure why it is needed here, but not in the workflow case.

Reply:
Because this needs to access the slot supplier to get the replacement for `heldSlotCount`. I could pass that through as a method on eagerWorker, or I guess have a reference to the atomic issuedCount in here, if that's better

Reply:
Yeah I would extend `eagerWorker` interface to provide the functionality you need. Eager activities used to be implemented like this and having different parts of the system reach into implementation details like this caused a number of bugs.